### ISSUE TITLE
Stream halts on order by over empty source

### ISSUE THREAD

**Issue by LaurensRietveld** [comunica/comunica#1091]
#### Issue type:

- :bug: Bug 

____
#### Description:

The node process terminates when reading an empty query bindings stream. What is particularly atypical:
- No error is thrown
- Subsequent lines are not executed
- The process exits with a `0` exit code

See [here](https://github.com/LaurensRietveld/comunica/blob/bug/executionStops/engines/query-sparql/lib/executionSeemsToStop.ts) for a minimum working example.

____
#### Environment:

```
| software         | version
| ---------------- | -------
| Comunica Engine  | 2.4.3 (dev)
| node             | v16.17.0
| npm              | 8.15.0
| yarn             | 1.22.19
| Operating System | linux (Linux 5.15.0-46-generic)

```

(note, the github issue template should be updated so that it references `node ./engines/query-sparql/bin/query.js -v` instead of `node ./packages/actor-init-query/bin/query.js -v`)

#### Crash log:

None

-----

## Bounty

A [bounty](https://comunica.dev/association/bounties/) has been placed on this issue by:

| |
|-|
| [<img alt="Comunica Association" src="https://comunica.dev/img/comunica_red.svg" height="50">](https://comunica.dev/association/) |
| €544 |

[Click here to learn more](https://comunica.dev/association/bounty_process/) if you're interested in claiming this bounty by resolving this issue.

**Comment by rubensworks** [comment]
Can confirm on my end.

Had to to replace the promise with the following though:
```
await new Promise(resolve => {
        results.on('data', () => { /* Force start reading */ });
        results.on('end', resolve);
    });
```

It looks like the problem goes away when removing the ORDER BY, so it seems likely that the problem is caused by the ORDER BY actor.

**Comment by jacoscaz** [comment]
Having worked on related issues in quadstore, I think it's worth pointing out that anything that re-orders a query result in memory has to buffer the entire result in memory. This can lead to errors and crashes _even in the absence of bugs_ if the result is big enough to make the process run out of memory. IMHO, there should be a configurable limit in the amount of items that ordering actors may buffer in memory, after which an explicit error should be thrown.

**Comment by rubensworks** [comment]
@LaurensRietveld We're looking into prioritizing this issue via the Comunica Association.
Are you interested in co-funding this as a bounty?

**Comment by rubensworks** [comment]
A bounty has been placed on this issue via the Comunica Association (see original post).

**Comment by rubensworks** [comment]
It looks like I already fixed this a while ago (2f4948e3ad5a0a5287f3afcaa731ad8113f30eaf). Fix is available since release 2.6.5.
